### ISSUE TITLE
why only 2 windows in tiling mode?

### ISSUE THREAD

**Issue by bazeeel** [codic12/worm#64]
Or something wrong at my end? when i press super + Return 3 times the third window is placed behind the other two

**Comment by codic12** [comment]
Did you toggle tiling mode with wormc layout tiling?

**Comment by bazeeel** [comment]
yes but only two window

**Comment by codic12** [comment]
I'm sorry, I don't really understand what you're saying. Can you send a video / screen recording as an example?

**Comment by is0n** [comment]
I have the same issue as seen here:

![recording](https://user-images.githubusercontent.com/57725322/164911079-18b8966b-905a-4cd8-8f96-bd70de9c23fb.gif)

When using the `tiling` layout, only 2 windows can be tiled per tag. If a third window is opened, all windows but the `master` window are hidden.

**Comment by bazeeel** [comment]
yes this is my problem too. Thanks  @is0n

**Comment by codic12** [comment]
I can reproduce the issue now. thanks. I'll try to find the commit where I introduced this bug and fix it.

**Comment by codic12** [comment]
Sorry for the problems. Looks like https://github.com/codic12/worm/pull/34 caused this issue, so I reverted that specific function ( tileWindows ) back to the version before refactor. Issue is solved in the Git version, and because this is a pretty serious bug I plan to make a release soon.

**Comment by is0n** [comment]
I can confirm this is working for me as shown below...

![demo](https://user-images.githubusercontent.com/57725322/165674066-384488d0-d1ce-4b64-9f85-7f878123b12a.gif)

I think it would be nicer if a window was focused after another window was closed but maybe that calls for a separate issue. 🤔

**Comment by codic12** [comment]
Btw, a minor thing but I noticed that the window buttons are mispositioned in your GIF. Does this and get fixed after a brief second, or only after forcing a re-render (which can be done by creating a new window, etc)?

> I think it would be nicer if a window was focused after another window was closed but maybe that calls for a separate issue. thinking
Indeed. I added this before, but reverted it because it didn't respect tags - I'll add a more full fledged implementation soon.

**Comment by codic12** [comment]
I doubt it, but can you check if that issue is fixed on master?

**Comment by is0n** [comment]
> can you check if that issue is fixed on master? Seems to work fine for me.

I've updated to the latest version and still no success. Can't say why it works on your system and not mine. Maybe @bazeeel can tell us if he has the same issue. 🤔

**Comment by codic12** [comment]
Here is a screen recording: are you doing something that I'm not here?


https://user-images.githubusercontent.com/48339289/165869689-d3d6fc1e-555b-4748-a496-7ecb45440428.mp4

**Comment by codic12** [comment]
Wait, can you try with other programs than alacritty, for example other terminal emulators? I think it may be doing some weird way of trying to resize itself after its tiled.

**Comment by is0n** [comment]
> Wait, can you try with other programs than alacritty, for example other terminal emulators?

I've tried it with `Firefox` and `Kitty` and I can confirm that the buttons are positioned correctly with those programs. Don't know why `Alacritty` poses an issue.

**Comment by codic12** [comment]
It probably uses _NET_MOVERESIZE_WINDOW, I will try to implement it in worm and then it should work.

**Comment by codic12** [comment]
(It doesn't, I just needed to add a renderTop to configure request and notify. You should be able to have it working now.

**Comment by is0n** [comment]
Works perfectly! Thanks again for your amazing work! 👍🏽

**Comment by bazeeel** [comment]
Yes!Now its working,thank you.
